src/winbind.c

```diff
@@ -49,28 +49,54 @@ static struct wbcContext *winbind_get_context(void)
     return ctx;
 }
 
-#else /* HAVE_PTHREAD */
+static long signal_ctx = 0;
```

Comment:
Could you pls add a comment here explaining the usage of signal_ctx ?
I.e. that address of this variable is used as a marker/indicator of single-threaded winbind context.

Can't we use 'NULL' here as a marker (instead of &signal_ctx )?

Reply:
As I was working at some point I need to differentiate between unallocated (NULL) and thread level context (&signal_ctx). Lemme see if I can simplify this one.

Reply:
actually can't for two reasons.
NULL is returned as error if allocating context fail in the "normal" case.
Also NULL is used to figure out if we need to request a context explicitly in external_get_creds() which can be called both from a ASC/ISC call where a context is available and other calls where preallocated contexts are not available and therefore need to be set up on the fly.